common/src/api/external/mod.rs

```diff
@@ -527,12 +527,21 @@ impl JsonSchema for RoleName {
 // in the database as an i64.  Constraining it here ensures that we can't fail
 // to serialize the value.
 //
-// TODO: custom JsonSchema and Deserialize impls to enforce i64::MAX limit
-#[derive(
-    Copy, Clone, Debug, Deserialize, Serialize, JsonSchema, PartialEq, Eq,
-)]
+// TODO: custom JsonSchema impl to describe i64::MAX limit; this is blocked by
+// https://github.com/oxidecomputer/typify/issues/589
```

Comment:
huh, that's surprising! This is still worth fixing, but to be clear, we'd rely on deserialize either way, so the behavior should be the "same" to clients afterwards, right? Would the only difference be "a more informative schema file"?

Reply:
Yeah, this PR fixes the server-side problem. The custom JSON schema would be so (e.g.) the CLI could validate the value even before sending the request.